planner/core/logical_plan_builder.go

```diff
@@ -299,8 +306,10 @@ func (b *PlanBuilder) buildJoin(joinNode *ast.Join) (LogicalPlan, error) {
 	switch joinNode.Tp {
 	case ast.LeftJoin:
 		joinPlan.JoinType = LeftOuterJoin
+		resetNotNullFlag(joinPlan.schema)
```

Comment:
we only need to reset the not null flag for the columns which come from the inner table.

Reply:
I tried to do so but It seems `LeftJoinKeys` is initialized during generating physical plan stage. We are touching logical plan generating logic.

Reply:
why it related to the `LeftJoinKeys`?

Reply:
I thought `LeftJoinKeys` is "columns which come from the inner table".